extensions/amp-user-notification/0.1/amp-user-notification.js

```diff
@@ -0,0 +1,335 @@
+    if (!this.activatedPromise_) {
+    // `layedoutPromise_` will block until there is no other
+    // `amp-user-notification` in display at this point in time.
+    this.activatedPromise_ = this.layedoutPromise_
+        .then(this.getAsyncCid_.bind(this))
+        .then(this.fetchShowEndpoint_.bind(this))
+        .then(this.onFetchShowEndpointSuccess_.bind(this))
+        .catch(this.onRejection_.bind(this));
+    }
+    return this.activatedPromise_;
+  }
+
+  /**
+   * Executes a `GET` request to the href given on the `data-show-if-href`
+   * attribute.
+   * @param {string} ampUserId
+   * @return {!Promise}
+   * @private
+   */
+  fetchShowEndpoint_(ampUserId) {
+    this.ampUserId_ = ampUserId;
+    const getReq = {
+      /** @type {GetRequestMetadata} */
+      body: {
+        elementId: this.elementId_,
+        ampUserId: ampUserId
+      }
+    };
+
+    return xhrFor(this.win_).fetchJson(this.showIfHref_, getReq);
+  }
+
+  /**
+   * Creates an POST to the specified `data-dismiss-href` url.
+   * @private
+   * @return {!Promise}
+   */
+  postDismissEnpoint_() {
+    return xhrFor(this.win_).fetchJson(this.dismissHref_, {
+      method: 'POST',
+      body: {
+        elementId: this.elementId_,
+        ampUserId: this.ampUserId_
+      }
+    });
+  }
+
+  /**
+   * Success handler for `fetchShowEndpoint_`.
+   * @param {!GetResponseMetadata}
+   * @return {!Promise|!GetResponseMetadata}
+   * @private
+   */
+  onFetchShowEndpointSuccess_(data) {
+    assert(typeof data.showNotification == 'boolean', '`showNotification` ' +
+        'should be a boolean. Got "%s" which is of type %s.',
+        data.showNotification, typeof data.showNotification);
+
+    if (data.showNotification) {
+      this.show();
+      return data;
+    }
+    // if no notification is shown, resolve right away.
+    return this.dialogResolve_();
+  }
+
+  /**
+   * Generic rejection handler for amp-user-notification.
+   * @private
+   */
+  onRejection_(msg) {
+    log.error(TAG_, 'An error ocurred on `amp-user-notification` ' +
+        `with html element id "${this.elementId_}". message: ${msg}`);
+  }
+
+  /**
+   * Get async cid service.
+   * @return {!Promise}
+   * @private
+   */
+  getAsyncCid_() {
+    return cidFor(this.win_).then(cid => {
+      // `amp-user-notification` is our cid scope, while we give it a resolved
+      // promise for the 2nd argument so that the 3rd argument (the
+      // persistentConsent) is the one used to resolve getting
+      // the external CID.
+      return cid.get('amp-user-notification',
+          Promise.resolve(), this.dialogPromise_);
+    });
+  }
+
+  /**
+   * Shows the current user notification. Adds the `.amp-active` class to
+   * the element.
+   * @export
+   */
+  show() {
+    this.element.classList.add('amp-active');
+  }
+
+  /**
+   * Hides the current user notification and invokes the `dialogResolve_`
+   * method. Removes the `.amp-active` class from the element.
+   * @export
```

Comment:
removed. this was previously so that the service could call show/dismiss and manage it.